Karma-Khameleons/bangularMobile
# Sam product pages

## Description
This initializes the product_list view and provides a first draft of productFactory.js.
productFactory will not work as intended at this point, as it is dependent upon session authentication, which is not yet in place. However, this should provide a rough outline for other team members to use for writing their own factories, which can be tweaked after we get session authentication in place.


## Number of Fixes
No real fixes, just some initialization of controllers/templates/factories.


## Related Ticket(s)
#2 


## Problem to Solve
Product information must be accessible/modifiable from the Angular app through the API.

## Changes
Describe the proposed changes, if any.


## Expected Behavior
When going to ```/product_detail``` or ```/product_list``` no CORS errors should be thrown, but a 403 (forbidden) error SHOULD be thrown.

## Steps to Test Solution
1. Run API on port ```localhost:8000```
1. Run angular app on port ```localhost:8080```
1. In your browser, visit ```/product_detail/1``` and ```/product_list``` to check that results align with Expected Behavior above
1. If you want to test further, make a new branch in your API repo, and on that branch comment out lines 44-47 of settings.py. 
```
'DEFAULT_PERMISSION_CLASSES': (
    'rest_framework.permissions.IsAuthenticated',
),
```
This will allow unauthenticated users (aka YOU) to access the data in the API.

**do not commit your changes in settings.py, those were for testing purposes only**

5. After commenting that out, the ```/product_list``` view, and all ```/product_detail/``` views linked to from the list view, should be visible

## Documentation
